### DISCUSSION TITLE
can't connect to localhost

### DISCUSSION THREAD

**Discussion by a2435191** [PrismarineJS/mineflayer#2069] [Q&A] [answered]
To preface, I have literally zero experience with Node or web stuff. I'm trying to run the following code on MacOS 1.15:
```js
const mineflayer = require('mineflayer')

const bot = mineflayer.createBot({
    host: 'localhost',
    port: '25565',
    username: 'email@example.com',
    hideErrors: false
})
```
But I get the following error:
```
Error: connect ECONNREFUSED 127.0.0.1:25565
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1148:16) {
  errno: -61,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 25565
}
```
Should I be configuring localhost to listen on 25565 somehow?

**Comment by Karang** [comment, marked as answer]
25565 is the default port of minecraft servers, if you changed it you need to set it to whatever port your server use.
Make sure you have a server running locally and that you can connect to it properly using your client.

**Reply by firewry** [reply]
i have set my port to my localhost port but when i run the file it changes it back to 25565

`Error: connect ECONNREFUSED 127.0.0.1:25565
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1157:16) {
  errno: -4078,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 25565
}`

**Reply by amoraschi** [reply]
I think you need to write the port as an integer, not as a string
